delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -38,71 +38,65 @@
 import java.time.temporal.ChronoUnit;
 import javax.annotation.Nullable;
 import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.MediaType;
 
 /**
  * Class to handle Api calls to System Service
  */
 public class AssessmentServiceClient {
 
-  ServiceDiscoverer serviceDiscoverer;
+  private final ServiceDiscoverer serviceDiscoverer;
+  private RetryPolicy retryPolicy;
+  private static final Gson GSON = new Gson();
 
   public AssessmentServiceClient(ServiceDiscoverer context) {
     this.serviceDiscoverer = context;
+    setUpRetryPolicy();
   }
 
-  public String retryableApiCall(String connectionUrl, String method, @Nullable String requestBody) {
+  public <T> T retryableApiCall(String connectionUrl, String method, @Nullable String requestBody,
+                                @Nullable Class<T> type) {
```

Comment:
type shouldn't be nullable. Also better for this to be a `Type` rather than `Class`